### ISSUE TITLE
Gradient of the DenseCRF loss

### ISSUE THREAD

**Issue by XiaoyanLi1** [meng-tang/rloss#3]
Hi,
I'm very interested at your work and want to follow your paper in ECCV2018. I notice that in the paper the DenseCRF loss is
![image](https://user-images.githubusercontent.com/10792053/50676658-73df4f00-1030-11e9-8d3c-3bf14820f3fb.png),
while in the code it is
![image](https://user-images.githubusercontent.com/10792053/50676839-8908ad80-1031-11e9-8f5a-b9d170740478.png)
In both the code and the paper, its loss is computed as 
![image](https://user-images.githubusercontent.com/10792053/50676720-d1739b80-1030-11e9-8ad5-0d154ec53a1d.png).
However, I think the gradient should be
![image](https://user-images.githubusercontent.com/10792053/50676761-10095600-1031-11e9-80ca-20e6f490efb7.png)
![image](https://user-images.githubusercontent.com/10792053/50676816-69718500-1031-11e9-8d09-68b4ace08eab.png).
Why is there a difference between the implementation and the theory? Should the first term of the gradient be ignored?

**Comment by Heyi007** [comment]
same doubt, do you figure it out ?

**Comment by meng-tang** [comment]
When summing over k for the first term, it becomes a constant no matter whether S is discrete or continuous. So I chose to ignore the first term.

**Comment by Tord-Zhang** [comment]
@meng-tang  Hi, I get confused. Since W is generated using Gaussian kernel, so every item in W is positive. S_k is the softmax output, so S_k is also positive. Then the gradient is always negative? So how can the gradient descent work?

**Comment by Tord-Zhang** [comment]
The loss just keep increasing......
